msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IBroker.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import java.util.Set;
+
+/**
+ * Used to define the basic set of methods that all Brokers must implement
+ *
+ * All methods are so they can be referenced by MSAL Java without an implementation, and by default simply throw an
+ * exception saying that a broker implementation is missing
+ */
+public interface IBroker {
+
+    /**
+     * Called when the 'useBroker' flag is first set, and will perform any initialization needed to make the broker available for future calls
+     *
+     * Will also detect dependency or accessibility errors with the Broker package, allowing an exception to provide the error to the app developer
+     * before any actual calls to a broker are made
+     */
+    default void initializeBroker(){
```

Comment:
Generally you don't need to put `Init` methods on interfaces. Implementations will use a ctor which can achieve this. 

In MSAL we expose a `IsInstalled` method, because there are cases where the broker is not installed. On Windows, this older versions like Win8 and Win Server 2016.

MSAL's logic is: 


```
Msal.AcquireToken(...) {
    if (broker.IsConfigured && broker.IsInstalled)  {
         return broker.AcquireToken();
    }
    
    return cache_or_aad.AcquireToken();
}
```

Reply:
Makes sense. Shouldn't these booleans be defined in IBroker interface? In .net IBroker interface, I see only 1 boolean IsBrokerInstalledAndInvokable. Is that the one to be used here? https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/Internal/Broker/IBroker.cs

Reply:
So 3 variables in IBroker interface - isAvailable, isConfigured and isInstalled? Are some of these redundant?

Reply:
No, just one, `IsAvailable` which means: 

- on Windows, it is a version of windows which has the broker 
- on Mac and Linux (which we don't integrate to right now), it'll mean "broker is installed"

"IsConfigured" is a setting on the PublicClientApplication object.